File: nova-api/src/main/java/xyz/xenondevs/nova/api/event/tileentity/TileEntityEvent.java
package xyz.xenondevs.nova.api.event.tileentity;

import org.bukkit.event.Event;
import org.jetbrains.annotations.NotNull;
import xyz.xenondevs.nova.api.tileentity.TileEntity;

/**
 * A {@link TileEntity} related event.
 */
public abstract class TileEntityEvent extends Event {
    
    private final @NotNull TileEntity tileEntity;
    
    public TileEntityEvent(@NotNull TileEntity tileEntity) {
        this.tileEntity = tileEntity;
    }
    
    public @NotNull TileEntity getTileEntity() {
        return tileEntity;
    }
    
    @Override
    public @NotNull String toString() {
        return "TileEntityEvent{" +
            "tileEntity=" + tileEntity +
            '}';
    }
    
}

File: Nova-API/src/main/kotlin/xyz/xenondevs/nova/api/event/tileentity/TileEntityEvent.kt
package xyz.xenondevs.nova.api.event.tileentity

import org.bukkit.event.Event
import xyz.xenondevs.nova.api.tileentity.TileEntity

/**
 * A [TileEntity] related event.
 */
abstract class TileEntityEvent(val tileEntity: TileEntity) : Event() {
    
    override fun toString(): String {
        return "TileEntityEvent(tileEntity=$tileEntity)"
    }
    
}